target/riscv: Comment that data1 might change.

timsifive opened the pull request:
In case in the future I have the same idea of optimizing progbuf writes again.

JanMatCodasip reviewed (commented):
I'm sorry but I believe this optimization is not legal w.r.t. the debug spec, and could make the memory write to stop woring on some hardware that is perfectly compliant.

The reason is that the debug spec does not guarantee `data1` retains its value after the execution of this abstract command, so we can't rely on it:

![image](https://github.com/riscv/riscv-openocd/assets/50193733/c5e8ce70-36a6-4bf6-80d3-9270d6174c72)

timsifive committed:
target/riscv: Comment that data1 might change.

In case in the future I have the same idea of optimizing progbuf writes
again.

Change-Id: Ie383487691cceeff75e2c22f4c85fc1fe4873937
Signed-off-by: Tim Newsome <tim@sifive.com>

timsifive commented:
Thanks for pointing that out. Changing this to a comment about why the optimization can't be done.
